pkg/webui/console/containers/device-onboarding-form/index.js

```diff
@@ -33,50 +44,159 @@ import DeviceTypeFormSection, {
 } from './device-type-form-section'
 import validationSchema from './validation-schema'
 
-const initialValues = merge({}, provisioningInitialValues, typeInitialValues)
+const initialValues = merge(
+  {
+    _registration: REGISTRATION_TYPES.SINGLE,
+  },
+  provisioningInitialValues,
+  typeInitialValues,
+)
 
 const DeviceOnboardingFormInner = () => {
   const {
-    values: {
-      frequency_plan_id,
-      lorawan_version,
-      lorawan_phy_version,
-      ids: { join_eui },
-    },
+    values: { frequency_plan_id, lorawan_version, lorawan_phy_version, ids },
   } = useFormContext()
+
   const maySubmit =
     Boolean(frequency_plan_id) &&
     Boolean(lorawan_version) &&
     Boolean(lorawan_phy_version) &&
-    join_eui.length === 16
+    ids?.join_eui?.length === 16
 
   return (
     <>
       <DeviceTypeFormSection />
       <DeviceProvisioningFormSection />
       {maySubmit && (
-        <SubmitBar>
-          <Form.Submit message={sharedMessages.addDevice} component={SubmitButton} />
-        </SubmitBar>
+        <>
+          <Form.Field title={m.afterRegistration} name="_registration" component={Radio.Group}>
+            <Radio label={m.singleRegistration} value={REGISTRATION_TYPES.SINGLE} />
+            <Radio label={m.multipleRegistration} value={REGISTRATION_TYPES.MULTIPLE} />
+          </Form.Field>
+          <SubmitBar>
+            <Form.Submit message={sharedMessages.addDevice} component={SubmitButton} />
+          </SubmitBar>
+        </>
       )}
     </>
   )
 }
 
 const DeviceOnboardingForm = () => {
+  const dispatch = useDispatch()
   const mayEditKeys = useSelector(state => checkFromState(mayEditApplicationDeviceKeys, state))
+  const appId = useSelector(state => selectSelectedApplicationId(state))
   const validationContext = React.useMemo(() => ({ mayEditKeys }), [mayEditKeys])
 
-  const handleSubmit = useCallback((values, formikBag, cleanedValues) => {
-    console.log(values, cleanedValues)
-    return Promise.resolve()
-  }, [])
+  const navigateToDevice = useCallback(
+    (appId, deviceId) => dispatch(push(`/applications/${appId}/devices/${deviceId}`)),
+    [dispatch],
+  )
+
+  const handleClaim = useCallback(
+    async (values, setSubmitting) => {
+      const { ids, authentication_code, qr_code } = values
+
+      let authenticatedIdentifiers
+      if (!qr_code) {
+        authenticatedIdentifiers = {
+          join_eui: ids.join_eui,
+          dev_eui: ids.dev_eui,
+          authentication_code,
+        }
+      }
+
+      try {
+        const device = await dispatch(
+          attachPromise(claimDevice(appId, qr_code, authenticatedIdentifiers)),
+        )
+        const { ids } = device
+        navigateToDevice(appId, ids.device_id)
+      } catch (error) {
+        setSubmitting(false)
```

Comment:
```suggestion
        setError(error)
```
The error needs to be stored and passed to the form. `setSubmitting()` is not required when the submit handler is asynchronous. We still unnecessarily use it in many submit handlers though.